pkg/analysis/passes/coderules/semgrep-rules.yaml

```
@@ -141,3 +141,20 @@ rules:
     message: "Console logging detected. Plugins should not log to the console."
     languages: [javascript, typescript]
     severity: WARNING
+
+  - id: detect-topnav-toggle
+    pattern-either:
+      - pattern: featureToggles.topnav
+    paths:
+      include:
+        - "src/**/*.ts"
+        - "src/**/*.tsx"
+      exclude:
+        - "*.spec.ts"
+        - "*.spec.tsx"
+        - "*.test.ts"
+        - "*.test.tsx"
+        - "*.js"
+    message: "The `topnav` toggle is deprecated and will be removed in a future version of Grafana. Plugins should default to using the code where the toggle is enabled."
+    languages: [javascript, typescript]
```

Nit: `*.js` files are excluded (`paths.exclude` property), so this should be:

```suggestion
    languages:
      - typescript
```